crates/typst/src/engine.rs

```diff
@@ -202,6 +327,61 @@ impl Sink {
     }
 }
 
+/// Checks if a given warning is suppressed given one span it has a tracepoint
+/// in. If one of the ancestors of the node where the warning occurred has a
+/// warning suppression decorator sibling right before it suppressing this
+/// particular warning, the warning is considered suppressed.
+fn check_warning_suppressed(
```

Comment:
I think `is_warning_suppressed` is a better name as it makes immediately clear what `true` means and what `false` means.